src/m_argv.c

```diff
@@ -69,6 +69,70 @@ boolean M_ParmExists(const char *check)
     return M_CheckParm(check) != 0;
 }
 
+#if defined(HAVE_GEN_PARAMS)
+#include "params.h"
+#include "i_system.h"
+
+static void CheckParams(int num)
+{
+  int i;
+
+  for (i = 0; i < arrlen(params); ++i)
+  {
+    if (!strcasecmp(myargv[num], params[i]))
+      return;
+  }
+
+  I_Error("No such option %s\n", myargv[num]);
+}
+
+static boolean CheckParamsWithArgs(int num)
+{
+  int i;
+
+  for (i = 0; i < arrlen(params_with_args); ++i)
+  {
+     if (!strcasecmp(myargv[num], params_with_args[i]))
+     {
+        if (num + 1 < myargc && myargv[num + 1][0] != '-')
+          return true;
+        else
+          I_Error("No parameter for %s", myargv[num]);
+     }
+  }
+
+  return false;
+}
+
+void M_CheckCommandLine(void)
+{
+  int i;
+
+  for (i = 1; i < myargc; i++)
+  {
+    if (CheckParamsWithArgs(i))
+    {
+      // -file and -warp may have multiple arguments
+      if (!strcasecmp(myargv[i], "-file") ||
```

Comment:
Does `-deh` also take multiple args?

Reply:
`-warp` can take one or two arguments, depending on the version. Not sure if we should check everything.

Reply:
Hm, "do it right or don't do it". 😁 Honestly, I'd expect command line checking code to detect that `-warp 1 2 3 4 5` cannot be right.

Reply:
> Hm, "do it right or don't do it".

Yes, that's the question 😄 How do you like this approach, is it worth finishing it?
There is also `-recordfrom` which only takes two arguments.

Reply:
BTW with `docgen` we can also generate code for `--help` options list, or maybe even `help <option>` messages.

Reply:
We will also need to check the validity of the parameter passed to e.g. the `-complevel` argument, else this whole checking procedure won't make too much sense.

Reply:
I think we should check the validity of `-complevel` and other arguments in the game code, such as the `-file` argument if the path/file is not found. There is 36 options with arguments and most of them silently ignore wrong input.